### ISSUE TITLE
ParallelDebugOutput only writes only part of the well information.

### ISSUE THREAD

**Issue by blattms** [OPM/opm-simulators#716]
While digging through the restart functionality and using that in parallel, I realized that in parallel only a part of the well information is written in [writeWells](/OPM/opm-simulators/blob/ccde26143c60a138f3100850dbd48e91d73312a9/opm/autodiff/ParallelDebugOutput.hpp#L462-L485), namely bhp and thp. The rest (temperature, perfrates, and wellrates) is missing, but I assume that it is needed for the restart and maybe even for plotting the wells.

**Comment by blattms** [comment]
I am pretty sure that this explains some of the wrong well plots (rates 0 for some wells) for model 2.

**Comment by blattms** [comment]
I guess the plots only need the wellrates, right?

**Comment by atgeirr** [comment]
I believe the answer is: it depends. On one hand the simulator should be able to run without getting the perforation rates or pressures, they are entirely dependent upon the other data (bhp, reservoir variables) so they should not be necessary for restart. Temperature is not used now. However, the "current control" setting can be important if we want the restarted simulator to behave exactly as if not restarted. Not sure if that is used though.

**Comment by blattms** [comment]
Thanks. As the output writer initializes a well manager with the global grid and EclipseState the current control should be ok.
Maybe we should open a feature request for the temperature, etc?

**Comment by atgeirr** [comment]
> As the output writer initializes a well manager with the global grid and EclipseState the current control should be ok

Not necessarily: by current control I do not mean the one requested in the deck schedule, but the one actually used (it can have switched due to breaking constraints). This information really has to come from the simulator.

**Comment by blattms** [comment]
Ok, resetting the bug tag.

**Comment by blattms** [comment]
Seems like once #836 will get merged this bug will bite us since the controls and perf_rates will be wrong.

**Comment by atgeirr** [comment]
> this bug will bite us

I assume you mean that the *parallell* case is buggy then?

**Comment by blattms** [comment]
> > this bug will bite us
> 
> I assume you mean that the *parallell* case is buggy then?

Yes.

**Comment by akva2** [comment]
@blattms i can include parallel but i need to know how to gather data (indexing). is there are global collection point ? as i understand it all i/o is manually serialized ?

**Comment by blattms** [comment]
The well data is gathered in [readWells](/OPM/opm-simulators/blob/ec955690cd0cd5606a322198d5db83656d75edf9/opm/autodiff/ParallelDebugOutput.hpp#L494-L521)/[writeWells](/OPM/opm-simulators/blob/ec955690cd0cd5606a322198d5db83656d75edf9/opm/autodiff/ParallelDebugOutput.hpp#L469-L492). We rely on the order of the wells as exported by EclipseParser::wells and the well information as exported by the eclipse schedule. As a wells is either completely on one process or not at all no further indexing should be needed.

**Comment by blattms** [comment]
@akva2 Will you work on this?

**Comment by akva2** [comment]
when things settle down i will look into it.

**Comment by andlaus** [comment]
is this still relevant?

**Comment by blattms** [comment]
Perfrates and perfpress are written with current master. Temperaturs is still not written.
